src/js/view/logo.js

```diff
@@ -40,18 +40,25 @@ export default function Logo(_model) {
         // apply styles onload when image width and height are known
         _img.onload = function () {
             // update logo style
-            var styles = {
+            const height = this.height;
+            const width = this.width;
+            const styles = {
                 backgroundImage: 'url("' + this.src + '")',
-                width: this.width,
-                height: this.height
+                width,
+                height
             };
             if (_settings.margin !== LogoDefaults.margin) {
-                var positions = (/(\w+)-(\w+)/).exec(_settings.position);
+                const positions = (/(\w+)-(\w+)/).exec(_settings.position);
                 if (positions.length === 3) {
                     styles['margin-' + positions[1]] = _settings.margin;
                     styles['margin-' + positions[2]] = _settings.margin;
                 }
             }
+
+            // Constraint logo size to 15% of their respective player dimension
+            styles.width = Math.min(width, Math.round(_model.get('containerWidth') / 6.67));
```

Comment:
`_model.get('containerWidth') * 0.15)`. Using 6.67 introduces unnecessary rounding.

Reply:
When the logo is larger than 15% of the player dimension, this calculation doesn't respect the aspect ratio of the logo.